src/lexer.cpp

```diff
@@ -0,0 +1,109 @@
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include "lexer.h"
+
+//VARIABLES
+std::vector<std::string> all_types = {"PROGRAM", "VAR", "END_VAR", "END_PROGRAM", "BOOL", "TRUE", "FALSE"}; //keywords
+std::vector<std::string> operators = {"<", ">", "+", "-", "%", ":="}; // operators
+std::vector<std::string> all_sc = {"{", "}", "[", "]", ";", ":"}; // special characters
+std::vector<Token> fin_tokens;
+
+void display(){
```

Comment:
if you'd like to make a function to pretty-print the token vector, it should take that vector as an argument rather than rely on the global variable.